src/unitxt/operators.py

```diff
@@ -1048,10 +1063,18 @@ def process(self, multi_stream: MultiStream) -> MultiStream:
             [*ele[0]] if isinstance(ele[0], tuple) else ele[0]
             for ele in values_and_counts
         ]
-        for name in multi_stream:
-            for instance in multi_stream[name]:
-                instance[self.to_field] = values_to_keep
-        return multi_stream
+
+        multi_stream_to_return = {}
+        addmostcommon_stream_operator = AddFieldValues(
+            fields_values={self.to_field: values_to_keep}
+        )
+        for stream_name, stream in multi_stream.items():
+            updated_stream = addmostcommon_stream_operator._process_single_stream(
+                stream
+            )
+            multi_stream_to_return[stream_name] = updated_stream
+
+        return MultiStream(multi_stream_to_return)
```

Comment:
```suggestion
        result_multi_stream = AddFieldValues(
            fields_values={self.to_field: values_to_keep}
        )
        return result_multi_stream
```
I think this could work

Reply:
Did not work:
Traceback (most recent call last):

```
  File "/home/dafna/workspaces/unitxt/tests/test_operators.py", line 691, in test_extract_values

    for k, v in output_multi_stream1.items():

AttributeError: 'AddFieldValues' object has no attribute 'items'
```

Reply:
Myself tried to copy all functions/methods from AddFieldValues and the functions invoked from them, copy all into ExtractFieldValues, by the word. And this did not work for me.  Looks deeper into Python objects and their properties that I could follow..

Reply:
sorry should be :
```python 
add_values = AddFieldValues(
            fields_values={self.to_field: values_to_keep}
        )
        result_multi_stream = add_values(original_multi_stream)
        return result_multi_stream
```
or similar